orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/job/WaitOnJobCompletion.groovy

```diff
@@ -120,8 +119,7 @@ public class WaitOnJobCompletion implements CloudProviderAware, OverridableTimeo
       }
 
       def name = names[0]
-      def parsedName = Names.parseName(name)
-      String appName = stage.context.moniker?.app ?: stage.context.application ?: parsedName.app
+      String appName = stage.context.moniker?.app ?: stage.context.application ?: stage.execution.application
```

Comment:
Should `parsedName` still be the last fallback here? It is strange to parse the job name as the application name, but I am worried that people may depend on this behavior and that changing it now would be a breaking change.

Reply:
that's right, the best way to fallback to the `parsedName` should be validate if there is an application with the `appName` if the application doesn't exist, then we will use the application name from `stage.execution.application`